app/src/main/java/com/example/universitieslist/view/UniversitiesScreen.kt

```diff
@@ -3,34 +3,65 @@ package com.example.universitieslist.view
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CircularProgressIndicator
 import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.example.universitieslist.R
 import com.example.universitieslist.view.components.AppToolbar
 import com.example.universitieslist.view.components.UniversityList
+import com.example.universitieslist.view.viewmodel.UniversityViewModel
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun UniversitiesScreen() {
+    val viewModel: UniversityViewModel = hiltViewModel()
+    val uiState by viewModel.uiState.collectAsState()
+    val snackBarHostState = remember { SnackbarHostState() }
+
+    LaunchedEffect(uiState) {
+        uiState.errorMessage?.let {
+            snackBarHostState.showSnackbar(it)
+        }
+    }
 
     Scaffold(
         topBar = {
             AppToolbar(
                 title = stringResource(R.string.app_name),
             )
         },
+        snackbarHost = { SnackbarHost(hostState = snackBarHostState) },
         content = { padding ->
             Box(
                 modifier = Modifier
                     .fillMaxSize()
                     .padding(padding),
                 contentAlignment = Alignment.TopStart
             ) {
-                UniversityList()
+                when {
+                    uiState.isLoading -> {
+                        CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+                    }
+
+                    else -> {
+                        uiState.universityList.let { universityList ->
+                            if (universityList != null) {
```

Comment:
Checking if _universityList_ is null or not could have been achieved using the `uiState.universityList?.let{ /**Your logic here*/ }`. This means if _universityList_ is null, the code block won't be executed.

Reply:
**Update:**
I just learnt something today: the difference between `?.let()` and `if(x != null)`. 
Basically, when there is no return value, the **traditional way** suits. Also, when the value being checked is mutable, the **Kotlin way** suits. 

I'm sorry for the mix-up. 🙏